File: SDKPackages/da/ios/Samples/KitchenSinkNew/KitchenSinkNew/TagPageView.m
/*******************************************************************************
 * Licensed Materials - Property of IBM
 * (C) Copyright IBM Corp. 2016
 * US Government Users Restricted Rights - Use, duplication or disclosure
 * restricted by GSA ADP Schedule Contract with IBM Corp.
 ******************************************************************************/
#import "TagPageView.h"

#import <UIKit/UIKit.h>
#import "DigitalAnalytics.h"

@interface TagPageView ()

@property (nonatomic,strong) NSString *pageName;
@property (nonatomic) BOOL sessionStarted;

@end

@implementation TagPageView

@synthesize pageName;
@synthesize sessionStarted;

-(id) initTag:(NSString *) pageNameParam sessionStarted:(BOOL) sessionStartedParam
{
    self = [super init];
    if(self)
    {
        self.pageName = pageNameParam;
        self.sessionStarted = sessionStartedParam;
    }
    return self;
}

-(void) executeTag
{
    if(self.sessionStarted)
    {
        NSLog(@"KitchenSinkNew is Starting New Digital Analytics Session");
        [DigitalAnalytics startNewSession];
    }
    
    BOOL success = [DigitalAnalytics firePageView:self.pageName category:@"iossdk" searchTerm:nil searchResult:nil attributes:nil cmmmc:nil];
    
    [self finish:success];
}
@end

File: SDKPackages/da/android/SampleCode/DigitalAnalyticsKitchenSink/app/src/main/java/com/tl/digitalanalytics/tagging/TagPageView.java
/*******************************************************************************
 * Licensed Materials - Property of IBM
 * (C) Copyright IBM Corp. 2016
 * US Government Users Restricted Rights - Use, duplication or disclosure
 * restricted by GSA ADP Schedule Contract with IBM Corp.
 ******************************************************************************/
package com.tl.digitalanalytics.tagging;

import android.app.Activity;
import android.util.Log;

import com.da.kitchensink.R;
import com.digitalanalytics.api.DigitalAnalytics;

/**
 * 
 * @author Sohil Shah (sohishah@us.ibm.com)
 * 
 */
public class TagPageView {
	private Activity activity;
	private boolean sessionStarted = false;

	public TagPageView(final Activity activity, boolean sessionStarted) {
		this.activity = activity;
		this.sessionStarted = sessionStarted;
	}

	public void executeTag() {
		try {
			Boolean success = true;
			
			// Do some session management
			if (this.sessionStarted) {
				success = DigitalAnalytics.startNewSession(this.activity);
			}

			// Perform Tagging
			String pageId = this.activity.getLocalClassName();
			String categoryId = "mobilesdk";
			success = DigitalAnalytics.firePageview(pageId, categoryId, null, null, null, null);

			this.finish(success);

		} catch (Exception e) {
			Log.e(Constants.LOG_TAG, e.getMessage(), e);
			this.finish(false);
		}
	}

	private void finish(boolean success) {
		// This is for functional testing/user feedback only. In practice,
		// tagging should just be silent
	    Helper.myAlertDialog(activity, success, null, R.string.tag_page_view_success, R.string.tag_page_view_failed);
	}
}
